app/components/ad-slot.js

```diff
@@ -66,6 +56,25 @@ export default Component.extend(
         },
         intersectionThreshold: 0,
       });
+
+      if (this.insertOnViewportEnter) {
+        return;
+      }
+
+      if (this.shouldWaitForUapResponse) {
+        ads.waitForUapResponse(
+          () => {
+            this.set('insertOnViewportEnter', true);
+          },
+          () => {
+            ads.pushSlotToQueue(name);
+          },
+        );
+
+        return;
+      }
+
+      ads.pushSlotToQueue(name);
```

Comment:
How about?

```js
didInsertElement() {
  this._super(...arguments);

  if (this.disableManualInsert) {
    return;
  }

  if (this.noAds) {
    return;
  }

  setProperties(this, {
    viewportTolerance: {
      top: 200,
      bottom: 200,
      left: 0,
      right: 0,
    },
    intersectionThreshold: 0,
  });

  this.insertAd();
},

/**
 * @private
 */
insertAd() {
  const ads = this.get('ads.module');
  const name = this.name;

  if (this.insertOnViewportEnter) {
    return;
  }

  if (this.shouldWaitForUapResponse) {
    ads.waitForUapResponse(
      () => this.set('insertOnViewportEnter', true),
      () => ads.pushSlotToQueue(name),
    );

    return;
  }

  ads.pushSlotToQueue(name);
},
```

`insertAd`, `pushAd`, `renderAd` - I am not entirely sure what name to choose.